e2e/stress/RaceConditionTest/Program.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            DeviceConnectionString = deviceConnectionString;
+        }
+
+        public async Task<int> RunTest_SendTenThousandMessagesAsync(TransportType transportType)
+        {
+            const int MaxExecutionTimeMinutes = 10;
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString, transportType);
+            deviceClient.OperationTimeoutInMilliseconds = 5;
+
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromMinutes(MaxExecutionTimeMinutes));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            deviceClient.RetryPolicy = RetryPolicyType.No_Retry;
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            List<Task> tasks = new List<Task>();
+
+            try
+            {
+                for (int i = 0; i < NumberOfParallelDeviceClientCalls; i++)
+                {
+                    Message message = new Message(Encoding.ASCII.GetBytes("Test"));
+                    tasks.Add(deviceClient.SendEventAsync(message, cancellationTokenSource.Token));
+                }
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return 1;
+            }
+
+            Console.WriteLine("Test passed (" + transportType + ")");
+            return 0;
+        }
+
+        public async Task<int> RunTest_SendThousandsOfMessagesAndCancelAsync(TransportType transportType)
+        {
+            const int MaxExecutionTimeInSeconds = 10;
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString, transportType);
+            deviceClient.OperationTimeoutInMilliseconds = 5;
+
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(MaxExecutionTimeInSeconds));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            deviceClient.RetryPolicy = RetryPolicyType.No_Retry;
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            List<Task> tasks = new List<Task>();
+
+            try
+            {
+                for (int i = 0; i < NumberOfParallelDeviceClientCalls; i++)
+                {
+                    Message message = new Message(Encoding.ASCII.GetBytes("Test"));
+                    tasks.Add(deviceClient.SendEventAsync(message, cancellationTokenSource.Token));
+                }
+
+                cancellationTokenSource.Cancel();
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return 1;
+            }
+
+            Console.WriteLine("Test passed (" + transportType + ")");
+            return 0;
+        }
+
+        public static int Main(string[] args)
+        {
```

Comment:
Why do you have a main in E2E ?

Reply:
This is 100% based on the other test already in the stress solution. I rather follow the same pattern so we don't loose focus on the purpose of testing this change.

Reply:
@prmathur-microsoft the VSTest harness is meant for unit tests mainly. Stress testing can cause the harness itself to behave abnormally. 
The folder maybe is not ideally named: e2e\stress. This is definitely not E2E but it does exercise public API.